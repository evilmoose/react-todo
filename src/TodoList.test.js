import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import TodoList from './TodoList'; // Adjust the import path according to your file structure

describe('TodoList', () => {
  it('renders without crashing', () => {
    render(<TodoList />);
  });

  it('initially displays the NewTodoForm', () => {
    render(<TodoList />);
    // Assuming the NewTodoForm has a distinctive element such as a form, input, or button that can be identified
    // This example assumes there's a button in the form. Adjust as necessary.
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  // Optionally, if you want to ensure the initial state has no todos rendered, you can check for the absence of certain elements.
  it('initially has no todos listed', () => {
    render(<TodoList />);
    // This assumes each todo would be in a 'li' element. Adjust the expectation if your structure is different.
    expect(screen.queryAllByRole('listitem')).toHaveLength(0);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<TodoList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
