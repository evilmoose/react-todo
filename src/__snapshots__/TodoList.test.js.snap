// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Add a todo!
      </button>
    </form>
  </div>
  <ul />
</div>
`;
